test/e2e/thanosruler_test.go

```diff
@@ -258,3 +259,90 @@ func testTRMinReadySeconds(t *testing.T) {
 		t.Fatalf("expected MinReadySeconds to be %d but got %d", updated, trSS.Spec.MinReadySeconds)
 	}
 }
+
+// Tests Thanos ruler -> Alertmanger path
+// This is done by creating a firing rule that will be picked up by
+// Thanos Ruler which will send it to Alertmanager, finally we will
+// use the Alertmanager API to validate that the alert is there
+func testTRAlertmanagerConfig(t *testing.T) {
+	const (
+		name       = "test"
+		group      = "thanos-alertmanager-test"
+		secretName = "thanos-ruler-alertmanagers-config"
+		configKey  = "alertmanagers.yaml"
+		testAlert  = "alert1"
+	)
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	// Create Alertmanager resource and service
+	alertmanager, err := framework.CreateAlertmanagerAndWaitUntilReady(context.Background(), ns, framework.MakeBasicAlertmanager(name, 3))
```

Comment:
A 1-replica Alertmanager is probably enough.
```suggestion
	alertmanager, err := framework.CreateAlertmanagerAndWaitUntilReady(context.Background(), ns, framework.MakeBasicAlertmanager(name, 1))
```